### PR TITLE
React ListItemTag

### PR DESCRIPTION
**Pull request by huayunh**
This is the same branch as the one mentioned in PR #100. I forget you can rename the branch to make the branch name legal -- and now I am too lazy to redo it.

Changes proposed in this Pull Request:
- added List Item Tag react 
- added those in storybook and showcase
- addressed comments mentioned in PR #100

### CODE REVIEW

File: react/components/src/core/ListItemTag/ListItemTag.test.tsx [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+    it('renders without crashing', () => {
+        const div = document.createElement('div');
+        ReactDOM.render(<ListItemTag label={'test'} />, div);
+        ReactDOM.unmountComponentAtNode(div);
+    });
+    it('should render with the list-item-tag class', () => {
+        const wrapper = shallow(<ListItemTag label={'test'} />);
+        expect(findByTestId('list-item-tag', wrapper)).toBeTruthy();
+    });
```
Comment (joebochill) [inline react/components/src/core/ListItemTag/ListItemTag.test.tsx]: the test name says it's checking for a class, but that's not what the test is checking...it should say:
```
it('should render list-item-tag wrapper')
```
or something to that effect.

File: react/components/src/core/ListItemTag/ListItemTag.tsx [hunk 2]
```diff
@@ -0,0 +1,75 @@
...
+
+export const ListItemTag: React.FC<ListItemTagProps> = (props: ListItemTagProps): JSX.Element => {
+    const {
+        classes: userClasses = {},
+        label,
+        fontColor,
+        backgroundColor,
+        variant,
+        style,
+        onClick,
```
Comment (joebochill) [inline react/components/src/core/ListItemTag/ListItemTag.tsx]:
```suggestion
```

File: react/components/src/core/ListItemTag/ListItemTag.tsx [hunk 3]
```diff
@@ -0,0 +1,75 @@
...
+    const defaultClasses = useStyles(theme);
+    const { root: rootUserClass, ...otherUserClasses } = userClasses;
+    return (
+        <Typography
+            classes={{ root: clsx(defaultClasses.root, rootUserClass), ...otherUserClasses }}
+            style={Object.assign(
+                {
+                    color: fontColor,
+                    backgroundColor: backgroundColor,
+                    cursor: onClick ? 'pointer' : 'default',
```
Comment (joebochill) [inline react/components/src/core/ListItemTag/ListItemTag.tsx]:
```suggestion
                    cursor: props.onClick ? 'pointer' : 'default',
```

File: react/components/src/core/ListItemTag/ListItemTag.tsx [hunk 4]
```diff
@@ -0,0 +1,75 @@
...
+            variant={variant || 'overline'}
+            noWrap={noWrap ? noWrap : true}
+            display={display || 'inline'}
+            data-test={'list-item-tag'}
+            onClick={(e): void => {
+                if (onClick) {
+                    onClick(e);
+                }
+                return;
+            }}
```
Comment (joebochill) [inline react/components/src/core/ListItemTag/ListItemTag.tsx]: onClick definition can be removed now with the above changes

File: react/docs/Hero.md [hunk 5]
```diff
@@ -48,7 +48,7 @@ import { Hero } from '@pxblue/react-components';
 The HeroBanner component is a simple wrapper component that is used to contain ```<Hero/>```s. It creates the flex container and sets up the spacing rules to display them. It accepts up to four ```<Hero/>``` components as its children.
```
Comment (joebochill) [inline react/docs/Hero.md]: change line above to HeroBanner (no space)

File: react/docs/Hero.md [hunk 6]
```diff
@@ -48,7 +48,7 @@ import { Hero } from '@pxblue/react-components';
...
 
 ### Hero Banner Usage
```
Comment (joebochill) [inline react/docs/Hero.md]: HeroBanner (no space)

File: react/docs/EmptyState.md [hunk 7]
```diff
@@ -1,13 +1,12 @@
-# Empty State Component
+# EmptyState
 The Empty State component is an element that can be used as a placeholder when no data is present (such as an empty list, or a placeholder page for future content). This is only used when no data is available, rather than during loading (see [empty states pattern](https://pxblue.github.io/patterns/empty-states)).
```
Comment (joebochill) [inline react/docs/EmptyState.md]: EmptyState (no space)

File: react/docs/InfoListItem.md [hunk 8]
```diff
@@ -1,10 +1,10 @@
-# Info List Item
+# InfoListItem
 The Info List Item is intended to be used in List views. It positions a title as well as optional subtitle(s), icon, and status stripe.
```
Comment (joebochill) [inline react/docs/InfoListItem.md]: InfoListItem (no space)

File: react/docs/ListItemTag.md [hunk 9]
```diff
@@ -0,0 +1,26 @@
...
+<img width="100%" alt="List Item Tag" src="./images/listItemTag.png">
+
+### Usage
+
+```typescript
+import { ListItemTag } from '@pxblue/react-components';
+
+// Simple usage passing props
+<ListItemTag label={"Foo Bar"} />
+
```
Comment (joebochill) [inline react/docs/ListItemTag.md]: Drop the basic example and just leave the one with the colors

File: react/scripts/buildTest.sh [hunk 10]
```diff
@@ -66,6 +66,12 @@ if [ ! -f ./DrawerLayout/DrawerLayout.js ]; then echo -e "${RED}Not Found${NC}"
 echo -ne " index.js:  "
 if [ ! -f ./DrawerLayout/DrawerLayout.js ]; then echo -e "${RED}Not Found${NC}" && exit 1; else echo -e "${GREEN}Found${NC}"; fi
 
+echo "Checking ListItemTag..."
+echo -ne " ListItemTag.js:  "
+if [ ! -f ./ListItemTag/ListItemTag.js ]; then echo -e "${RED}Not Found${NC}" && exit 1; else echo -e "${GREEN}Found${NC}"; fi
+echo -ne " index.js:  "
+if [ ! -f ./ListItemTag/index.js ]; then echo -e "${RED}Not Found${NC}" && exit 1; else echo -e "${GREEN}Found${NC}"; fi
+
```
Comment (joebochill) [inline react/scripts/buildTest.sh]: Will you add checks for the Spacer component here if I forgot to?

Comment (huayunh) [reply react/scripts/buildTest.sh]: you did in the Spacer, which already got merged into dev

**Comment by joebochill** [comment]
Can we also add `prop-types` definitions so we don't have to go back and do it later

File: react/scripts/buildTest.sh (hunk 10 above)
Comment (joebochill) [reply react/scripts/buildTest.sh]: okay cool, I couldn't remember if I did or not

**Pull request merged**